pkg/resourcetotelemetry/resource_to_telemetry.go

```diff
@@ -83,6 +83,10 @@ func addAttributesToMetric(metric *pdata.Metric, labelMap pdata.AttributeMap) {
 		addAttributesToNumberDataPoints(metric.Sum().DataPoints(), labelMap)
 	case pdata.MetricDataTypeHistogram:
 		addAttributesToHistogramDataPoints(metric.Histogram().DataPoints(), labelMap)
+	case pdata.MetricDataTypeSummary:
+		addAttributesToSummaryDataPoints(metric.Summary().DataPoints(), labelMap)
+	case pdata.MetricDataTypeExponentialHistogram:
+		addAttributesToExponentialHistogramDataPoints(metric.ExponentialHistogram().DataPoints(), labelMap)
 	}
```

Comment:
Is there any linter that we can enable to check this? I know is hard since there is no enum type, but maybe some structure tags can help. This needs to be investigated because I bet there are more places like this.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7532